### PR TITLE
Limit visibility of mxnet output symbols to consumers of the library

### PR DESCRIPTION
**Pull request by tdhd**
*Description of changes:*

* Move functionality from `mxnet_output_symbols.py` to private class `_MXNetModule`
* Make use of `_MXNetModule` from `Imputer`
* Fixed some missing parameters in docstrings

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: datawig/imputer.py [hunk 1]
```diff
@@ -311,37 +311,9 @@ def checkpoint(epoch, sym, arg, aux):
...
-            mx.sym.Group([loss] + output_symbols),
-            context=self.ctx,
-            data_names=[name for name, dim in iter_train.provide_data],
-            label_names=[name for name, dim in iter_train.provide_label],
-            fixed_param_names=fixed_params
-        )
-
-        self.module.bind(data_shapes=iter_train.provide_data, label_shapes=iter_train.provide_label)
+    def __build_module(self, iter_train: ImputerIterDf) -> mx.mod.Module:
+        mod = _MXNetModule(self.ctx, self.label_encoders, self.data_featurizers, self.final_fc_hidden_units)
```
Comment (felixbiessmann) [inline datawig/imputer.py]: is ``_MXNetModule.__init__`` always used with immediately calling ``.get`` afterwards? then we could consider passing the iterator directly to ``__init__``?

File: datawig/imputer.py [hunk 2]
```diff
@@ -962,3 +887,210 @@ def calibrate(self, test_iter: ImputerIterDf):
...
+        :param label_encoders: list of label column encoders
+        :param data_featurizers: list of data featurizers
+        :param final_fc_hidden_units: list of number of hidden parameters
+        """
+        self.ctx = ctx
+        self.data_featurizers = data_featurizers
+        self.label_encoders = label_encoders
+        self.final_fc_hidden_units = final_fc_hidden_units
+
+    def get(self, iter_train: ImputerIterDf) -> mx.mod.Module:
```
[the remark this answers is not in the document]
Comment (tdhd) [reply datawig/imputer.py]: good idea, changed it

File: datawig/imputer.py [hunk 3]
```diff
@@ -40,7 +39,7 @@
 from . import calibration
 
 
-class Imputer:
+class Imputer(object):
```
[the remark this answers is not in the document]
Comment (tdhd) [reply datawig/imputer.py]: yes, just looked it up, in python3 there is no need to inherit from `object` anymore

File: datawig/imputer.py (hunk 1 above)
Comment (tdhd) [reply datawig/imputer.py]: not sure, it can be confusing to create an instance of `_MXNetModule` but return a `mxnet.mod.Module`. What do you think?

**Pull request merged**